crates/ruff_linter/src/source_kind.rs

```diff
@@ -234,9 +235,19 @@ impl std::fmt::Display for CodeDiff<'_> {
             // individual lines
             for change in hunk.iter_changes() {
                 match change.tag() {
-                    ChangeTag::Equal => write!(f, " {}", change.value())?,
-                    ChangeTag::Delete => write!(f, "{}{}", "-".red(), change.value().red())?,
-                    ChangeTag::Insert => write!(f, "{}{}", "+".green(), change.value().green())?,
+                    ChangeTag::Equal => write!(f, " {}", change.value().show_nonprinting())?,
+                    ChangeTag::Delete => write!(
+                        f,
+                        "{}{}",
+                        "-".red(),
+                        change.value().show_nonprinting().red()
+                    )?,
+                    ChangeTag::Insert => write!(
+                        f,
+                        "{}{}",
+                        "+".green(),
+                        change.value().show_nonprinting().green()
+                    )?,
```

Comment:
Nit, I would introduce a variable with the cleaned-up code to reduce the diff size and reduce the risk that one path misses the call.

```rust
let change = change.value().show_nonprinting();

match change.tag {
	ChangeTag::Equal => write!(f, " {}", change)?,
	...
}
```